sdk/core/core-amqp/src/retry.ts

```diff
@@ -194,3 +231,97 @@ export async function retry<T>(config: RetryConfig<T>): Promise<T> {
     throw lastError;
   }
 }
+
+/**
+ *
+ * Utility to exponentially retry on the given operation for until the specified maximum
+ * delay between operations is reached.
+ *
+ * @param {RetryConfig<T>} config Parameters to configure retry operation.
+ *
+ * @return {Promise<T>} Promise<T>.
+ */
+export async function exponentialRetry<T>(config: RetryConfig<T>): Promise<T> {
+  validateRetryConfig(config);
+  if (config.maxExponentialRetryDelayInMs == undefined || config.maxExponentialRetryDelayInMs < 0) {
+    config.maxExponentialRetryDelayInMs = defaultMaxDelayForExponentialRetryInMs;
+  }
+  if (config.minExponentialRetryDelayInMs == undefined || config.minExponentialRetryDelayInMs < 0) {
+    config.minExponentialRetryDelayInMs = defaultMinDelayForExponentialRetryInMs;
+  }
+
+  let lastError: MessagingError | undefined;
+  let result: any;
+  let success = false;
+
+  // Based on specified max delay, the total number of attempts to be made can be determined as,
+  // minExponentialRetryDelayInMilliseconds + 2^(retryAttempts) should be less than or equal to maxExponentialRetryDelayInMilliseconds
+  const totalNumberOfAttempts = Math.floor(
+    Math.log2(config.maxExponentialRetryDelayInMs - config.minExponentialRetryDelayInMs)
```

Comment:
Why is `totalNumberOfAttempts` being calculated using delays?
This should be `maxRetries` + 1

The delay related options should only govern the delay between retries